### PR TITLE
Init container explorer model

### PR DESCRIPTION
**Pull request by jdneo**
This PR add the model for Container Explorer.

The OkHttpClient is used to send the HTTP requests. And the usage of RESTful API is referenced to [Azure CLI's implementation](https://github.com/Azure/azure-cli/blob/096dbb966c4bb7bf8a0674fc44387b6543ee35be/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py#L191)

### CODE REVIEW

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+     */
+    public List<String> listRepositories(@NotNull String serverUrl, @NotNull String username,
+                                         @NotNull String password) throws Exception {
+        List<String> ret = new ArrayList<>();
+        OkHttpClient client = createRestClient(username, password);
+        final String host = URL_PREFIX + serverUrl;
+        String path = REPOSITORY_PATH;
+        boolean executeNextHttpCall = true;
+        while (executeNextHttpCall) {
+            HttpUrl route = HttpUrl.parse(host + path);
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: exception throw in this method?

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * list repositories under the given private registry.
+     */
+    public List<String> listRepositories(@NotNull String serverUrl, @NotNull String username,
+                                         @NotNull String password) throws Exception {
+        List<String> ret = new ArrayList<>();
+        OkHttpClient client = createRestClient(username, password);
+        final String host = URL_PREFIX + serverUrl;
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: find api to construct url path

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        String path = REPOSITORY_PATH;
+        boolean executeNextHttpCall = true;
+        while (executeNextHttpCall) {
+            HttpUrl route = HttpUrl.parse(host + path);
+            Request request = new Request.Builder().url(route).get().build();
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: check route null before next step?

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            try (Response response = client.newCall(request).execute()) {
+                if (response.isSuccessful()) {
+                    Gson gson = new Gson();
+                    Repository repo = gson.fromJson(response.body().string(), Repository.class);
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: change Repository name

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                    ret.addAll(repo.getRepositories());
+                    // TODO: add page division
+                    String linkHeader = response.header(HEADER_LINK);
+                    if (!Utils.isEmptyString(linkHeader)) {
+                        path = linkHeader.substring(linkHeader.indexOf("<") + 1, linkHeader.lastIndexOf(">"));
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: be care about the string index

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java (hunk 4 above)
Comment (jdneo) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: Changed to Catalog

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java (hunk 3 above)
Comment (jdneo) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: Yes

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+import okhttp3.Credentials;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class ContainerExplorerMvpModel {
+
+    private static final OkHttpClient sharedClient = new OkHttpClient();
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/container/ContainerExplorerMvpModel.java]: not static

**Pull request merged**